std-bits/base/src/main/java/org/enso/base/parser/NegativeSign.java

```diff
@@ -0,0 +1,40 @@
+package org.enso.base.parser;
+
+import java.util.Optional;
+
+public enum NegativeSign {
+  /** No sign encountered, so could be either. */
+  UNKNOWN,
+  /** Minus or Plus sign - e.g. +123 or -123. */
+  MINUS,
+  /** Brackets - e.g. (123) */
+  BRACKET_OPEN;
+
+  /**
+   * Checks if the given character is a valid negative sign.
+   *
+   * @param c the character to check
+   * @return the next negative sign state, or null if the character is invalid
```

Comment:
the doc is inconsistent with return value - says it returns null but actually `Optional.empty()` is returned.